services/datamanager/builtin/capture_test.go

```diff
@@ -139,6 +139,8 @@ func TestDataCaptureEnabled(t *testing.T) {
 				AssociatedAttributes: associations,
 			})
 			test.That(t, err, test.ShouldBeNil)
+			b := dmsvc.(*builtIn)
+			test.That(t, b.propagateDataSyncConfig(), test.ShouldBeNil)
```

Comment:
In order to not rewrite the tests, I call `propagateDataSyncConfig` after every reconfigure which will happen at latest 1 second after the previous reconfigure in production.